src/MongoDB.Driver.Core/Core/Operations/RetryabilityHelper.cs

```diff
@@ -0,0 +1,98 @@
+﻿/* Copyright 2018-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+
+namespace MongoDB.Driver.Core.Operations
+{
+    internal static class RetryabilityHelper
+    {
+        // private static fields
+        private static readonly HashSet<Type> __resumableChangeStreamExceptions;
+        private static readonly HashSet<ServerErrorCode> __resumableChangeStreamErrorCodes;
+        private static readonly HashSet<Type> __retryableWriteExceptions;
+        private static readonly HashSet<ServerErrorCode> __retryableWriteErrorCodes;
+
+        // static constructor
+        static RetryabilityHelper()
+        {
+            var resumableAndRetryableExceptions = new HashSet<Type>()
+            {
+                typeof(MongoConnectionException),
+                typeof(MongoNotPrimaryException)
+            };
+
+            __resumableChangeStreamExceptions = new HashSet<Type>(resumableAndRetryableExceptions)
+            {
+                typeof(MongoCursorNotFoundException)
+            };
+
+            __retryableWriteExceptions = new HashSet<Type>(resumableAndRetryableExceptions)
+            {
+                typeof(MongoNodeIsRecoveringException)
```

Comment:
I can't think of a reason why this wouldn't also be retryable for change streams.  Also, found this in the server source code:

                uassert(ErrorCodes::NotMasterOrSecondary,
                        "node is recovering",
                        !replCoord->getMemberState().recovering());

So the driver is already translating ServerErrorCode.NotMasterOrSecondary to MongoNodeIsRecoveringException based on the error string, which is on the list of resumable error code, and then not treating it as resumable after the translation.